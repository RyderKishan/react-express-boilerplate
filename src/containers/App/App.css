body {
  font-family: 'Futura BK BT', Arial, sans-serif !important;
  font-size: 14px;
  margin: 0;
  height: 100vh;
  width: 100vw;
  max-width: 2560px;
  max-height: 1440px;
}

#app {
  height: 100%;
  width: 100%;
}

.App {
  height: 100%;
  width: 100%;
}

header {
  height: 70px;
  background: #0D475E;
  padding: 0 16px;
  color: white;
}

article {
  height: calc(100% - 70px);
  display: flex;
  overflow: auto;
}

@font-face {
  font-family: "Futura BK BT";
  src: url("/public/fonts/Futura.ttf");
}