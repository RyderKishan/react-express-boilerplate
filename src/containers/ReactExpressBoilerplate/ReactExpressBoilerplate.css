body {
  font-family: 'Futura BK BT', Arial, sans-serif !important;
  font-size: 1em;
  margin: 0;
  height: 100vh;
  width: 100vw;
  max-width: 2560px;
  max-height: 1440px;
}

#app {
  height: 100%;
  width: 100%;
}

.ReactExpressBoilerplate {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
}

header {
  height: 5em;
  background: #0D475E;
  padding: 0 1em;
  color: white;
}

footer {
  height: 3em;
  color: white;
  background: #0D475E;
  display: flex;
  justify-content: space-around;
  align-items: center;
  font-size: 11px;
}

article {
  flex: 1;
  overflow: auto;
}

@font-face {
  font-family: "Futura BK BT";
  src: url("/public/fonts/Futura.ttf");
}