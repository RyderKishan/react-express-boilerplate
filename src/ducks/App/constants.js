const GET_USER_DETAILS = 'App/GET_USER_DETAILS';
const SET_USER_DETAILS = 'App/SET_USER_DETAILS';

export default {
  GET_USER_DETAILS,
  SET_USER_DETAILS,
};
