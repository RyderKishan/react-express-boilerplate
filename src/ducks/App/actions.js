import * as Constants from './constants';

const getUserDetails = () => ({
  type: Constants.GET_USER_DETAILS,
});

const setUserDetails = (userDetails) => ({
  type: Constants.SET_USER_DETAILS,
  userDetails,
});

export {
  getUserDetails,
  setUserDetails,
};
