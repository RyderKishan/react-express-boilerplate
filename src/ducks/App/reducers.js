import * as Constants from './constants';

const AppReducer = (state = {}, action) => {
  switch (action.type) {
    case Constants.SET_USER_DETAILS:
      return {
        ...state,
        userDetails: action.userDetails,
      };

    default:
      return state;
  }
};

export default AppReducer;
