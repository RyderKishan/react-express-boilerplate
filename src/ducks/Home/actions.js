import * as Constants from './constants';

const getPosts = () => ({
  type: Constants.GET_POSTS,
});

const setPosts = (posts) => ({
  type: Constants.SET_POSTS,
  posts,
});

export {
  getPosts,
  setPosts,
};
