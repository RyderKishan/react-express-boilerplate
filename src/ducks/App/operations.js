import * as Actions from './actions';

const dispatchGetUserDetails = (dispatch) => () => {
  dispatch(Actions.getUserDetails());
};

const dispatchSetUserDetails = (dispatch) => (userDetails) => {
  dispatch(Actions.setUserDetails(userDetails));
};

export default {
  dispatchGetUserDetails,
  dispatchSetUserDetails,
};
