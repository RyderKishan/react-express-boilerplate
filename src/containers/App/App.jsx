import React, { useEffect, shallowEqual } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { withRouter, Route, Switch } from 'react-router-dom';

import * as Actions from '../../ducks/App/actions';
import * as Selectors from '../../ducks/App/selectors';

import './App.css';
import Home from '../Home';
import Header from '../../components/Header';
import FallBack from '../../components/FallBack';

const App = () => {
  const dispatch = useDispatch();
  const getUserDetails = dispatch(Actions.getUserDetails());
  const {
    userDetails,
  } = useSelector((state) => ({
    userDetails: Selectors.userDetails(state),
  }),
  shallowEqual);
  useEffect(() => {
    getUserDetails();
  }, []);
  return (
    <div className="App">
      <header>
        <Header appName="React Webpack" userDetails={userDetails} />
      </header>
      <article className="body">
        <Switch>
          <Route path="/" exact render={() => <Home />} />
          <Route path="/contact" render={() => <div className="Contact">Contact</div>} />
          <Route path="/profile" render={() => <div className="Profile">Profile</div>} />
          <Route component={FallBack} />
        </Switch>
      </article>
    </div>
  );
};

export default withRouter(App);
