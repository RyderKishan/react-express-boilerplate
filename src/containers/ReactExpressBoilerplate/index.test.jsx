import React from 'react';
import renderer from 'react-test-renderer';
import { render, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import ReactExpressBoilerplate from '.';
import * as Actions from '../../ducks/App/actions';
import Mocks from '../../../api/stubs/mocks';

const mockDispatch = jest.fn();
mockDispatch.mockReturnValue(jest.fn());

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => ({
    userDetails: Mocks.userDetails,
  }),
}));

describe('<ReactExpressBoilerplate /> - Hooks Test', () => {
  const wrapper = mount(
    <MemoryRouter>
      <ReactExpressBoilerplate />
    </MemoryRouter>,
  );
  it('useEffect', () => {
    expect(wrapper.find('.ReactExpressBoilerplate').exists()).toBe(true);
    expect(mockDispatch).toHaveBeenCalledWith(Actions.getUserDetails());
  });
});

describe('<ReactExpressBoilerplate /> - Snapshot Tests Renderer', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <ReactExpressBoilerplate />
    </MemoryRouter>,
  );
  it('Snapshot renderer', () => {
    expect(tree).toMatchSnapshot();
  });
});

const linkRenderer = (link) => render(
  <MemoryRouter initialEntries={[link]}>
    <ReactExpressBoilerplate />
  </MemoryRouter>,
);

describe('<ReactExpressBoilerplate /> - Route Tests', () => {
  it('Render Home Container', () => {
    const wrapper = linkRenderer('/');
    expect(wrapper.find('article')).toHaveLength(1);
    expect(wrapper.find('.Home')).toHaveLength(1);
  });
  it('Render Contact Container', () => {
    const wrapper = linkRenderer('/contact');
    expect(wrapper.find('article')).toHaveLength(1);
    expect(wrapper.find('.Contact')).toHaveLength(1);
  });
  it('Render About Container', () => {
    const wrapper = linkRenderer('/about');
    expect(wrapper.find('article')).toHaveLength(1);
    expect(wrapper.find('.About')).toHaveLength(1);
  });
  it('Render FallBack Container', () => {
    const wrapper = linkRenderer('/error');
    expect(wrapper.find('article')).toHaveLength(1);
    expect(wrapper.find('.FallBack')).toHaveLength(1);
  });
});
