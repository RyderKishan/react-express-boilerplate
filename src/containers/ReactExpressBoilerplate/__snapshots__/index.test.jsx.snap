// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactExpressBoilerplate /> - Snapshot Tests Renderer Snapshot renderer 1`] = `
<div
  className="ReactExpressBoilerplate"
>
  <header>
    <div
      className="Header"
    >
      <div
        className="image-container"
      >
        <img
          alt="logo"
          onClick={[Function]}
          role="none"
          src="/public/images/gk_tag.png"
        />
      </div>
      <div
        className="loginUser"
      >
        <div
          className="routing"
        >
          <span
            onClick={[Function]}
            role="none"
          >
            About
          </span>
          <span
            onClick={[Function]}
            role="none"
          >
            Contact Us
          </span>
        </div>
        <div
          className="search"
        >
          <div
            className="Search"
          >
            <div
              className="searchArea"
            >
              <input
                className="REB-input"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Search"
                required={true}
                size={null}
                type="search"
              />
            </div>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              onClick={[Function]}
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
        </div>
        <span>
          Login User
        </span>
        <div
          className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          L
        </div>
      </div>
    </div>
  </header>
  <article
    className="body"
  >
    <div
      className="Home"
    >
      <div
        className="actions"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          id="fetch-results"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Fetch Results
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          id="clear-results"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear Results
          </span>
        </button>
      </div>
      <div
        className="results"
      >
        <div
          className="postContainer"
        >
          <div
            className="make-center"
          >
            No Results
          </div>
        </div>
      </div>
    </div>
    <footer>
      C @ Balkishan S
    </footer>
  </article>
</div>
`;
